agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -569,6 +574,14 @@ static Configuration getConfigurationFromConfigFile(Path configPath, boolean str
     }
   }
 
+  // Function to remove all non-ASCII characters
+  // Note: Is public, since we have to use this function in tests.
+  public static String cleanJsonContent(String text) {
+    // strips off all non-ASCII characters
+    text = text.replaceAll("[^\\x00-\\x7F]", "");
```

Comment:
One issue with this implementation is this function will remove any non ascii characters present in the json config values itself. But I am assuming our config will always have ascii characters.